fe/fe-core/src/main/java/org/apache/doris/datasource/ExternalCatalog.java

```diff
@@ -291,6 +291,7 @@ public void setUninitialized(boolean invalidCache) {
         this.objectCreated = false;
         this.initialized = false;
         this.invalidCacheInInit = invalidCache;
+        onClose();
```

Comment:
This should be called only when Catalog is dropped.
For refresh, we need another method